ui/packages/atlasmap/src/UI/dialogs/ParametersDialog.tsx

```diff
@@ -0,0 +1,178 @@
+import React, {
+  FunctionComponent,
+  useState,
+  useCallback, useEffect,
+} from "react";
+
+import {
+  Button,
+  Form,
+  FormGroup, FormSelect, FormSelectOption, Split, SplitItem,
+  TextInput,
+} from "@patternfly/react-core";
+
+import {
+  ConfirmationDialog,
+  IConfirmationDialogProps,
+} from "./ConfirmationDialog";
+import {PlusIcon, TimesIcon} from "@patternfly/react-icons";
+
+export interface IParameter {
+  name: string;
+  value: string;
+  options?: IParameterOption[];
+  hidden?: boolean;
+  hideable?: boolean;
```

Comment:
I'm a bit confused by these two parameters. What do you think about renaming `hidden`  to `required` flipping the logic in the code, and dropping `hideable` completely by making all not required options to be removable (since this is the way it's set up anyway?)

Reply:
From the perspective of this specific use case it's enough to have one `required` parameter. It's a bit more flexibility to be also able to specify whether to hide a parameter or not when first showing the form without making it required.

Reply:
Changed to use only `required` parameter.